godtools/App/Share/Data/UserDetailsRepository/Cache/RealmUserDetailsCache.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  RealmUserDetailsCache.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 11/21/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+import Combine
+
+class RealmUserDetailsCache {
+    
+    private let realmDatabase: RealmDatabase
+    private let userDetailsSync: RealmUserDetailsCacheSync
+    private let authTokenRepository: MobileContentAuthTokenRepository
+    
+    init(realmDatabase: RealmDatabase, userDetailsSync: RealmUserDetailsCacheSync, authTokenRepository: MobileContentAuthTokenRepository) {
+        
+        self.realmDatabase = realmDatabase
+        self.userDetailsSync = userDetailsSync
+        self.authTokenRepository = authTokenRepository
+    }
+    
+    func getUserDetailsChanged() -> AnyPublisher<Void, Never> {
+        
+        return realmDatabase.openRealm().objects(RealmUserDetails.self)
+            .objectWillChange
+            .eraseToAnyPublisher()
+    }
+    
+    func getUserDetails() -> UserDetailsDataModel? {
```

Comment:
Hey @rachaelblue I think this method could be a little more descriptive since it's accessing authTokenRepository to fetch the authorized userId.  Something like ```getAuthUserDetails()```.